aic-emu/AicEmu.cc

```diff
@@ -0,0 +1,140 @@
+    AicSocketData_t socketInfo;
+};
+
+void Usage()
+{
+    std::cout << "Usage: aic-emu <mandatory options> [non-mandatory options] " << std::endl;
+    std::cout << "\nMandatory options:" << std::endl;
+    std::cout << "  --cmd <yml file path>  : Path of YML file containing AiC cmd sequence" << std::endl;
+    std::cout << "  --content <clip path>  : Path of clip to serve as content" << std::endl;
+    std::cout << "  --hwc-sock <socketpath>: Path of Unix Socket file" << std::endl;
+    std::cout << "\nNon-Mandatory options:" << std::endl;
+    std::cout << "  --device <device path> : Device path. Default /dev/dri/renderD128" << std::endl;
+    std::cout << "  --help, -h             : Print this help and exit" << std::endl;
+    return;
+}
+
+int ParseArgs(AicConfigData_t& config, int argc, char** argv)
+{
+    int idx;
+    for (idx = 1; idx < argc; ++idx)
+    {
+        if (std::string("-h") == argv[idx] || std::string("--help") == argv[idx])
+        {
+            Usage();
+            exit(0);
+        }
+        else if (std::string("--cmd") == argv[idx])
+        {
+            if (++idx >= argc)
+                break;
+            config.ymlFileName = std::string(argv[idx]);
+        }
+        else if (std::string("--content") == argv[idx])
+        {
+            if (++idx >= argc)
+                break;
+            config.contentFileName = std::string(argv[idx]);
+        }
+        else if (std::string("--device") == argv[idx])
+        {
+            if (++idx >= argc)
+                break;
+            config.deviceString = std::string(argv[idx]);
+        }
+        else if (std::string("--hwc-sock") == argv[idx])
+        {
+            if (++idx >= argc)
+                break;
+            config.socketInfo.hwc_sock = std::string(argv[idx]);
+
+            //Make rest of fields 0
+            config.socketInfo.session_id = 0;
+            config.socketInfo.user_id = 0;
+            config.socketInfo.android_instance_id = 0;
+        }
+        else
+        {
+            break;
+        }
+    }
+
+    if (config.ymlFileName.empty())
+    {
+        std::cout << "Specification of YMLfile with AiC commands is mandatory" <<std::endl;
+        Usage();
+        return AICS_ERR_INVALID_ARGS;
+    }
+
+    if (config.contentFileName.empty())
+    {
+        std::cout << "Specification of Content file is mandatory" <<std::endl;
+        Usage();
+        return AICS_ERR_INVALID_ARGS;
+    }
+
+    if (config.socketInfo.hwc_sock.empty())
+    {
+        std::cout << "Specification of Unix Socket file path is mandatory" <<std::endl;
+        Usage();
+        return AICS_ERR_INVALID_ARGS;
+    }
+
+    return AICS_ERR_NONE;
+}
+
+int main(int argc, char** argv)
+{
+    int status = AICS_ERR_NONE;
+
+    AicConfigData_t config;
+    status = ParseArgs(config, argc, argv);
+    CHECK_STATUS(status);
+
+    CmdHandler* handler = new CmdHandler(config.ymlFileName, config.contentFileName);
+
+    status = handler->InitYmlParser();
```

Comment:
what's the reason to have these 3 init steps exposed to end-user explicitly? Why you can't do that within constructor? or within single init function?

Reply:
No good reason actually: just the way the code evolved. moved to within a single init function.